### PR TITLE
feat(rust/advance-chunks): rework with new API deisgn

### PR DESCRIPTION
**Pull request by hyf0**
### Description

refer: https://github.com/rolldown/rolldown/discussions/2118

### CODE REVIEW

**Comment by hyf0** [comment]
* **#2152** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2152?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2146** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2146?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hyf0 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/rolldown/src/stages/generate_stage/code_splitting.rs [hunk 1]
```diff
@@ -295,4 +278,76 @@ impl<'a> GenerateStage<'a> {
...
+      return;
+    }
+
+    // Higher priority group goes first.
+    match_groups.sort_by_key(|group| Reverse(group.priority));
+
+    let mut name_to_module_group: FxHashMap<ArcStr, ModuleGroup> = FxHashMap::default();
+
+    for normal_module in self.link_output.module_table.modules.iter().filter_map(Module::as_ecma) {
+      if !normal_module.is_included {
```
Comment (underfin) [inline crates/rolldown/src/stages/generate_stage/code_splitting.rs:312]: Here lost 
``` .rs
if module_to_assigned[normal_module.idx] {
      continue;
      }
```

File: crates/rolldown/src/stages/generate_stage/code_splitting.rs [hunk 2]
```diff
@@ -295,4 +278,76 @@ impl<'a> GenerateStage<'a> {
...
+    else {
+      return;
+    };
+
+    if match_groups.is_empty() {
+      return;
+    }
+
+    // Higher priority group goes first.
+    match_groups.sort_by_key(|group| Reverse(group.priority));
```
Comment (underfin) [inline crates/rolldown/src/stages/generate_stage/code_splitting.rs:307]: Not sure if two groups `[a, b]` has same priority, look like we should take `a` first?

File: crates/rolldown/src/stages/generate_stage/code_splitting.rs (hunk 1 above)
Comment (hyf0) [reply crates/rolldown/src/stages/generate_stage/code_splitting.rs:312]: no need. A module won't show twice here

File: crates/rolldown/src/stages/generate_stage/code_splitting.rs (hunk 2 above)
Comment (hyf0) [reply crates/rolldown/src/stages/generate_stage/code_splitting.rs:307]: Notice this is a stable sort. If the priority is the same, then we just pick the first one.

**Comment by IWANABETHATGUY** [comment]
Do you mean `rework` in title?

**Comment by hyf0** [comment]
> Do you mean `rework` in title?

Yeah.

**Pull request merged**